src/scripts/migrate.py

```diff
@@ -52,10 +50,20 @@
     'mondo_id': 'ID', 'mondo_label': 'LABEL', 'xref': 'A oboInOwl:hasDbXref',
     'xref_source': '>A oboInOwl:source SPLIT=|', 'original_label': '', 'definition': 'A IAO:0000115', 'parents': 'SC %'}
 
+def _valid_parent_conditions(
+    parents: List[CURIE], mapped: Set[CURIE], excluded: Set[CURIE], obsolete: Set[CURIE]
+) -> bool:
+    """This is an optional, stricter check on slurp candidacy / order.
+
+    For a term to be immediately migratable, it must either (a) have no parents, or (b) all of its parents must be
+    mapped, obsolete, or excluded"""
+    return not parents or [any([x in y for y in [mapped, excluded, obsolete]]) for x in parents]
```

Comment:
I can only hope this expression is correct, as I am not really able to mentally parse it :D

Reply:
Lol, well it is supposed to do what is written in the comment above it. The `any()` part also translates to:

`[x if (x in mapped) or (x in excluded) or (x in obsolete) for x in parents]`

`mapped`, `excluded`, and `obsolete` are lists of terms.